lib/ics_renderer.rb

```diff
@@ -10,27 +10,25 @@ def render
     output << "METHOD:PUBLISH\r\n"
     output << "PRODID:-//uk.gov/GOVUK calendars//EN\r\n"
     output << "CALSCALE:GREGORIAN\r\n"
-    @events.each_with_index do |event, i|
-      output << render_event(event, i)
-    end
+    @events.each { |event| output << render_event(event) }
     output << "END:VCALENDAR\r\n"
   end
 
-  def render_event(event, sequence)
+  def render_event(event)
     output =  "BEGIN:VEVENT\r\n"
     # The end date is defined as non-inclusive in the RFC (2445 section 4.6.1)
     output << "DTEND;VALUE=DATE:#{(event.date + 1.day).strftime('%Y%m%d')}\r\n"
     output << "DTSTART;VALUE=DATE:#{event.date.strftime('%Y%m%d')}\r\n"
     output << "SUMMARY:#{event.title}\r\n"
-    output << "UID:#{uid(sequence)}\r\n"
+    output << "UID:#{uid}\r\n"
     output << "SEQUENCE:0\r\n"
     output << "DTSTAMP:#{dtstamp}\r\n"
     output << "END:VEVENT\r\n"
   end
 
-  def uid(sequence)
+  def uid
     @path_hash ||= Digest::MD5.hexdigest(@cal_path)
-    "#{@path_hash}-#{sequence}@gov.uk"
+    "#{@path_hash}-#{SecureRandom.uuid}@gov.uk"
```

Comment:
I think we'll want the the ID to remain the same for the same events, otherwise each time an external tool scrapes our calendar it'll think all the events have been deleted and replaced with new ones.

I suggest perhaps generating a hash of the date associated with the event (assuming all our dates are unique, which I think they are):

```ruby
hash = Digest::SHA1.hexdigest(event.date.iso8601)
```

Reply:
Or alternatively, if we want to stick with UUIDs, we could put them directly into the data files and generate them as we add new dates in.

Reply:
What are the restrictions on the UID format?  Would `#{event.date.strftime('%Y%m%d')}-#{event.title}` work?

Reply:
Both good points, thanks. 😉